spec/posthog/feature_flag_spec.rb

```diff
@@ -0,0 +1,3087 @@
+      assert_requested :post, 'https://app.posthog.com/decide/?v=2', times: 1
+    end
+
+    it 'evaluates flag with complex definition' do
+      api_feature_flag_res = {
+        "flags": [
+          {
+            "id": 1,
+            "name": "Beta Feature",
+            "key": "complex-flag",
+            "is_simple_flag": false,
+            "active": true,
+            "filters": {
+                "groups": [
+                    {
+                        "properties": [
+                            {
+                                "key": "region",
+                                "operator": "exact",
+                                "value": ["USA"],
+                                "type": "person",
+                            },
+                            {
+                                "key": "name",
+                                "operator": "exact",
+                                "value": ["Aloha"],
+                                "type": "person",
+                            },
+                        ],
+                        "rollout_percentage": 100,
+                    },
+                    {
+                        "properties": [
+                            {
+                                "key": "email",
+                                "operator": "exact",
+                                "value": ["a@b.com", "b@c.com"],
+                                "type": "person",
+                            },
+                        ],
+                        "rollout_percentage": 30,
+                    },
+                    {
+                        "properties": [
+                            {
+                                "key": "doesnt_matter",
+                                "operator": "exact",
+                                "value": ["1", "2"],
+                                "type": "person",
+                            },
+                        ],
+                        "rollout_percentage": 0,
+                    },
+                ],
+            }
+        },]
+      }
+      stub_request(
+        :get,
+        'https://app.posthog.com/api/feature_flag/local_evaluation?token=testsecret'
+      ).to_return(status: 200, body: api_feature_flag_res.to_json)
+
+      stub_request(:post, 'https://app.posthog.com/decide/?v=2')
+      .to_return(status: 200, body: {"featureFlags": {"complex-flag": "decide-fallback-value"}}.to_json)
+
+
+      c = Client.new(api_key: API_KEY, personal_api_key: API_KEY, test_mode: true)
+
+      expect(c.get_feature_flag("complex-flag", "some-distinct-id", person_properties: {"region" => "USA", "name" => "Aloha"})).to eq(true)
+      expect(c.get_feature_flag("complex-flag", "some-distinct-id", person_properties: {"region": "USA", "name": "Aloha"})).to eq(true)
```

Comment:
sorry if this is addressed somewhere else already but how come we're checking for this twice? Just to make sure the hash syntax is working?

Reply:
yep exactly, it tests that both symbol & string hashes work.

Reply:
(not just syntax, as they represent different things in ruby, and we want the library to accept both)